### PR TITLE
Engine - Add filter by namespaces in test mode

### PR DESCRIPTION
**Pull request by NahuFigueroa97**
|Related issue|
|---|
|#18523|

## Description

In this PR a solution is proposed to filter the traces in the output of the mode using namespaces. For this, the parameter --namespaceid was added in the test run command.

### CODE REVIEW

File: src/engine/source/api/include/api/test/handlers.hpp [hunk 1]
```diff
@@ -233,7 +233,8 @@ api::Handler sessionsGet(const std::shared_ptr<api::sessionManager::SessionManag
  * @return api::Handler
  */
 api::Handler runPost(const std::shared_ptr<api::sessionManager::SessionManager>& sessionManager,
-                     const std::shared_ptr<::router::Router>& router);
+                     const std::shared_ptr<::router::Router>& router,
+                     const std::shared_ptr<store::IStore>& store);
```
Comment (soynof) [inline src/engine/source/api/include/api/test/handlers.hpp:237]: Please, check documentation.

File: src/engine/source/proto/src/test.proto [hunk 2]
```diff
@@ -112,8 +112,9 @@ message RunPost_Request
     optional string protocol_queue = 3;       // The protocol queue for the event.
     optional DebugMode debug_mode = 4; // Allows you to choose between obtaining only the output or the output plus the
                                        // traces with or without verbosity.
-    optional string asset_trace = 5;   // Asset of which you only want to have details
+    repeated string asset_trace = 5;   // Asset of which you only want to have details
     optional string protocol_location = 6; // The protocol location for the event.
+    repeated string namespaceid = 7;       // Namespaces where are the assets to test
```
Comment (juliancnn) [inline src/engine/source/proto/src/test.proto:117]:
```suggestion
    repeated string namespaces = 7;       // Namespaces where are the assets to test
```

**Review by juliancnn — CHANGES_REQUESTED** [review]

File: src/engine/source/router/include/router/policyManager.hpp [hunk 3]
```diff
@@ -115,13 +115,33 @@ class PolicyManager
      * @param traceCallback The callback function to be invoked with the generated trace data.
      * @param name Name of the policy for which output and trace are to be obtained.
      * @param instance Instance of the policy for which output and trace are to be obtained.
+     * @param assets Vector of asset names to subscribe to.
```
Comment (soynof) [inline src/engine/source/router/include/router/policyManager.hpp:118]: Check parameters order

File: src/engine/source/router/include/router/router.hpp [hunk 4]
```diff
@@ -204,12 +204,30 @@ class Router
      * @param outputCallback The callback function to be invoked with the generated output data.
      * @param traceCallback The callback function to be invoked with the generated trace data.
      * @param name Name of the policy for which output and trace are to be obtained.
+     * @param assets Vector of asset names to subscribe to.
```
Comment (soynof) [inline src/engine/source/router/include/router/router.hpp:207]: Check parameters order

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by JavierBejMen — APPROVED** [review]
LGTM

File: src/engine/source/router/include/router/router.hpp (hunk 4 above)
Comment (JcabreraC) [reply src/engine/source/router/include/router/router.hpp:206]: done

File: src/engine/source/router/include/router/policyManager.hpp (hunk 3 above)
Comment (JcabreraC) [reply src/engine/source/router/include/router/policyManager.hpp]: Done

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**